app/audit/models/EnterClientUTRAuditModel.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package audit.models
+
+import play.api.libs.json.{JsObject, JsValue, Json}
+
+case class EnterClientUTRAuditModel(isSuccessful: Boolean, nino: String, mtditid: String, arn: Option[String], saUtr: String, credId: Option[String]) extends ExtendedAuditModel {
+
+  override val transactionName: String = enums.TransactionName.EnterClientUTR
+
+  override val auditType: String = enums.AuditType.EnterClientUTR
+
+  private val successOutcome: JsObject = Json.obj(
+    "outcome" ->
+      Json.obj("isSuccessful" -> isSuccessful)
+  )
+  private val failureOutcome: JsObject = Json.obj(
+    "outcome" ->
+      Json.obj(
+        "isSuccessful" -> isSuccessful,
+        "failureCategory" -> "API_FAILURE",
+        "failureReason" -> "API returned error - unable to login agent"
+      )
+  )
+
+  private val outcome: JsObject = if (isSuccessful) successOutcome else failureOutcome
+
+  private val userDetailsJson: JsObject = Json.obj(
+    "nino" -> nino,
+    "mtditid" -> mtditid,
+    "agentReferenceNumber" -> arn,
```

Comment:
:thinking: could we use ++ syntax for Optional fields (arn and credId) so that they are removed from the audit if empty?